src/librustc_resolve/lib.rs

```diff
@@ -3512,6 +3634,59 @@ fn path_names_to_string(path: &Path, depth: usize) -> String {
     names_to_string(&names[..])
 }
 
+/// When an entity with a given name is not available in scope, we search for
+/// entities with that name in all crates. This method allows outputting the
+/// results of this search in a programmer-friendly way
+fn show_candidates(session: &mut DiagnosticBuilder,
+                   span: &syntax::codemap::Span,
```

Comment:
`Span` is `Copy`, so there's no need for the reference -- this can just be `span: syntax::codemap::Span,`